src/Mvc/Mvc.ViewFeatures/src/RazorComponents/HtmlRenderer.cs

```diff
@@ -247,6 +259,13 @@ private async Task<(int, ArrayRange<RenderTreeFrame>)> CreateInitialRenderAsync(
 
             return (componentId, GetCurrentRenderTreeFrames(componentId));
         }
+
+        private class HtmlRenderingContext
```

Comment:
struct?

Reply:
It's only one per render, and if we make it a mutable struct we have to be careful about passing by ref everywhere. I'd be happier keeping it as `class` for simplicity.